### DISCUSSION TITLE
Suggested approach to generate chart image data from prompt

### DISCUSSION THREAD

**Discussion by eeadie** [microsoft/semantic-kernel#8513] [General]
Hi,

I was wondering if there is any recommendations for returning an image as a response to a users prompt? For example, perhaps returning a chart to display data as opposed to a textual response. I have spotted the ITextToImageService but I am not sure if that would be the best approach or if I should create a plugin that has responsibility for generating chart images:

https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.Abstractions/AI/TextToImage/ITextToImageService.cs

Thanks,

Euan

**Comment by moonbox3** [comment]
Hi @eeadie, this can be handled using OpenAI Assistants. You can have a look at this sample for inspiration: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/Agents/OpenAIAssistant_ChartMaker.cs. Would this help your use-case? Please let us know. Thanks.
